docs/reference/commandline/dockerd.md

```diff
@@ -1349,7 +1349,7 @@ This is a full example of the allowed configuration options on Linux:
   "cluster-store-opts": {},
   "containerd": "/run/containerd/containerd.sock",
   "containerd-namespace": "docker",
-  "containerd-plugin-namespace": "docker-plugins",
+  "containerd-plugins-namespace": "docker-plugins",
```

Comment:
Looking at the source-code, the option in JSON looks to be named `containerd-plugin-namespace` (singular "plugin"); https://github.com/moby/moby/blob/7b9275c0da707b030e62c96b679a976f31f929d3/daemon/config/config.go#L266

```go
ContainerdPluginNamespace string `json:"containerd-plugin-namespace,omitempty"`
```


What error did you get when using this option in your daemon.json?

Reply:
(Note that the _flag_ looks to be using plural (which may be somewhat confusing); I'll have a look at adding a hidden flag to support both options; https://github.com/moby/moby/blob/7b9275c0da707b030e62c96b679a976f31f929d3/cmd/dockerd/config.go#L100)

Reply:
Hello, I am using Docker and Docker where are the daemon configuration is loaded onto the D in D image.

When I use the non plural version, I get the error that I am sharing below:

<img width="1076" alt="Screenshot 2021-04-22 at 15 26 49" src="https://user-images.githubusercontent.com/4193588/115731777-50667c80-a37f-11eb-91ac-dd0e514449e6.png">

(With configuration panel removed):

<img width="1196" alt="Screenshot 2021-04-22 at 15 26 55" src="https://user-images.githubusercontent.com/4193588/115731864-63794c80-a37f-11eb-9fe4-1d4f0dc667da.png">

When I do use the plural version however, the DND container stat as I would expect:

<img width="1096" alt="Screenshot 2021-04-22 at 15 24 19" src="https://user-images.githubusercontent.com/4193588/115731261-e77f0480-a37e-11eb-8b00-7bfc30eee124.png">

(dont worry about those logs concerning missing containers; the point is the docker daemon is running).

Reply:
For more context, this is the `docker:20.10.6-dind` image.